src/main/java/io/logz/jmx2logzio/listener/prometheus/PromLogzioSender.kt

```diff
@@ -0,0 +1,128 @@
+package io.logz.jmx2logzio.listener.prometheus
+
+import io.logz.prometheus.protobuf.Types
+import io.logz.sender.HttpsRequestConfiguration
+import io.logz.sender.SenderStatusReporter
+import io.logz.sender.exceptions.LogzioServerErrorException
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+import java.util.concurrent.atomic.AtomicBoolean
+
+class PromLogzioSender(
+    private val reporter: SenderStatusReporter,
+    private val tasksExecutor: ScheduledExecutorService,
+    private val timeSeriesQueue: TimeSeriesDiskQueue,
+    private val debug: Boolean = false,
+    private val drainTimeoutSec: Long = 5,
+    httpsRequestConfiguration: HttpsRequestConfiguration
+) {
+    companion object {
+        private const val MAX_SIZE_IN_BYTES = 3 * 1024 * 1024 // 3 MB
+        private const val FINAL_DRAIN_TIMEOUT_SEC = 20
+    }
+
+    private val promHttpsSyncSender = PromHttpsSyncSender(httpsRequestConfiguration, reporter)
+    private val drainRunning = AtomicBoolean(false)
+
+    init {
+        debug("Created new LogzioPrometheusSender class")
+    }
+
+    fun start() {
+        tasksExecutor.scheduleAtFixedRate(this::drainQueueAndSend, 0, drainTimeoutSec, TimeUnit.SECONDS)
+    }
+
+    fun stop() {
+        // Creating a scheduled executor, outside of logback to try and drain the queue one last time
+        val executorService = Executors.newSingleThreadExecutor()
+        debug("Got stop request, Submitting a final drain queue task to drain before shutdown. Will timeout in $FINAL_DRAIN_TIMEOUT_SEC seconds.")
+
+        try {
+            executorService.submit { this.drainQueue() }[FINAL_DRAIN_TIMEOUT_SEC.toLong(), TimeUnit.SECONDS]
+        } catch (e: InterruptedException) {
+            debug("Waited $FINAL_DRAIN_TIMEOUT_SEC seconds, but could not finish draining. quitting.", e)
+        } catch (e: ExecutionException) {
+            debug("Waited $FINAL_DRAIN_TIMEOUT_SEC seconds, but could not finish draining. quitting.", e)
+        } catch (e: TimeoutException) {
+            debug("Waited $FINAL_DRAIN_TIMEOUT_SEC seconds, but could not finish draining. quitting.", e)
+        } finally {
+            executorService.shutdownNow()
+        }
+    }
+
+    fun send(message: ByteArray) {
```

Comment:
You are: PromLogzioSender: Accept TimeSeries type, and internally convert to ByteArray. The user isn't to blame you use inside BigQueue